src/main/java/mil/dds/anet/resources/EntityAvatarResource.java

```diff
@@ -0,0 +1,116 @@
+  }
+
+  /**
+   * Gets the avatar for this relatedObject uuid, null if none existing
+   *
+   * @param relatedObjectUuid the relatedObjectUuid
+   * @return the entity avatar
+   */
+  @GraphQLQuery(name = "entityAvatar")
+  public EntityAvatar getEntityAvatar(@GraphQLRootContext Map<String, Object> context,
+      @GraphQLArgument(name = "relatedObjectType") String relatedObjectType,
+      @GraphQLArgument(name = "relatedObjectUuid") String relatedObjectUuid) {
+    return entityAvatarDao.getByRelatedObject(relatedObjectType, relatedObjectUuid).orElse(null);
+  }
+
+  /**
+   * Creates or updates an entity avatar
+   * 
+   * @param context the context
+   * @param entityAvatar the entity avatar
+   * @return the entity avatar
+   */
+  @GraphQLMutation(name = "createOrUpdateEntityAvatar")
+  public Integer createOrUpdateEntityAvatar(@GraphQLRootContext Map<String, Object> context,
+      @GraphQLArgument(name = "entityAvatar") EntityAvatar entityAvatar) {
+    final int numRows;
+    final Person user = DaoUtils.getUserFromContext(context);
+
+    // Check if user is authorized to manipulate avatars for this relatedObjectUuid
+    assertPermission(user, entityAvatar.getRelatedObjectType(),
+        entityAvatar.getRelatedObjectUuid());
+
+    // Do we have an avatar already for this entity?
+    final Optional<EntityAvatar> dbEntityAvatar = entityAvatarDao.getByRelatedObject(
+        entityAvatar.getRelatedObjectType(), entityAvatar.getRelatedObjectUuid());
+
+    if (dbEntityAvatar.isEmpty()) {
+      numRows = entityAvatarDao.insert(entityAvatar);
+      AnetAuditLogger.log("Avatar for entity {} created by {}", entityAvatar.getRelatedObjectUuid(),
+          user);
+    } else {
+      numRows = entityAvatarDao.update(entityAvatar);
+      AnetAuditLogger.log("Avatar for entity {} updated by {}", entityAvatar.getRelatedObjectUuid(),
+          user);
+    }
+
+    // GraphQL mutations *have* to return something, so we return the number of updated rows
+    return numRows;
+  }
+
+  /**
+   * Deletes the entity avatar if any
+   * 
+   * @param context the context
+   * @param relatedObjectType the relatedObjectType
+   * @param relatedObjectUuid the relatedObjectUuid
+   * @return the number of rows deleted
+   */
+  @GraphQLMutation(name = "deleteEntityAvatar")
+  public Integer deleteEntityAvatar(@GraphQLRootContext Map<String, Object> context,
+      @GraphQLArgument(name = "relatedObjectType") String relatedObjectType,
+      @GraphQLArgument(name = "relatedObjectUuid") String relatedObjectUuid) {
+    final Person user = DaoUtils.getUserFromContext(context);
+
+    // Check if user is authorized to manipulate avatars for this relatedObjectUuid
+    assertPermission(user, relatedObjectType, relatedObjectUuid);
+
+    int numRows = entityAvatarDao.delete(relatedObjectType, relatedObjectUuid);
+    if (numRows > 1) {
+      AnetAuditLogger.log("Avatar for entity {} deleted by {}", relatedObjectUuid, user);
```

Comment:
Please also log the entity type (prevents having to guess which table was updated).